doc/developer/design/20220819_bounded_input_reliance.md

```diff
@@ -0,0 +1,164 @@
+# Bounded Input Reliance
+
+**Note**: This design document relies on changes in [this pr], which is expected
+to merge soon.
+
+[this pr]: https://github.com/MaterializeInc/materialize/pull/14355
+
+## Summary
+
+In the [source ingestion pipeline design doc] required Materialize to guarantee _bounded input reliance_.
+This means that sources should not re-read data that has already been read and written to persist.
```

Comment:
Nit but it doesn't need to mean this. It only needs to mean that we are able to bound the information we need to re-read not that we cannot reread data that are written to persist.

In either case, I would strike "read and".